src/Yoti/Entity/MultiValue.php

```diff
@@ -0,0 +1,190 @@
+<?php
+namespace Yoti\Entity;
+
+class MultiValue extends \ArrayObject
+{
+    /**
+     * @var array Original unfiltered Items.
+     */
+    private $originalItems = [];
+
+    /**
+     * @var boolean Flag to make MultiValue immutable.
+     */
+    private $isImmutable = false;
+
+    /**
+     * @var callable[] Filter callbacks.
+     */
+    private $filters = [];
+
+    /**
+     * MultiValue Constructor.
+     *
+     * @param array $items
+     */
+    public function __construct($items)
+    {
+        $this->originalItems = array_values($items);
+        parent::__construct($this->originalItems);
+    }
+
+    /**
+     * Filters Items using callback.
+     *
+     * @param callable $callback
+     * @return MultiValue
+     */
+    public function filter($callback)
+    {
+        $this->assertMutable('Attempting to filter immutable array');
+        $this->filters[] = $callback;
+        $this->applyFilters();
+        return $this;
+    }
+
+    /**
+     * Apply all filters.
+     */
+    private function applyFilters()
+    {
+        // Only apply filters if this is mutable.
+        if ($this->isImmutable) {
+            return;
+        }
+
+        // Reset to original items before filtering.
+        $this->exchangeArray($this->originalItems);
+
+        // Apply filter to items.
+        if (count($this->filters) > 0) {
+            $filtered_array = array_filter(
+                $this->getArrayCopy(),
+                [$this, 'filterItem']
```

Comment:
Is this reflective call necessary?
Can't we replace this with something that is checked at load time?

Reply:
I can change this to an [anonymous function / closure](http://php.net/manual/en/functions.anonymous.php)? It might make more sense in this case

Reply:
I've changed this in 7f0912e0e594ad6f031b4bef55ea0ba4c7d7f1bf